Sources/UI/Camera/CameraService.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  CameraService.swift
+//  PovioKit
+//
+//  Created by Ndriqim Nagavci on 13/10/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import AVFoundation.AVMediaFormat
+import PovioKit
+
+protocol CameraServiceProtocol: AnyObject {
+  func requestCameraAuthorization(_ completion: ((Bool) -> Void)?)
+  func authorizationStatus(forType mediaType: AVMediaType) -> AVAuthorizationStatus
+  func isCameraAvailable(position: AVCaptureDevice.Position) -> Bool
+}
+
+class CameraService { /* see extension bellow for implementation */ }
+
+// MARK: - CameraService Protocol
+extension CameraService: CameraServiceProtocol {
+  /// Check if app is authorized to use camera or not. For the first time this method will ask user for permission.
+  func requestCameraAuthorization(_ completion: ((Bool) -> Void)?) {
+    let mediaType = AVMediaType.video
+    let authStatus = authorizationStatus(forType: mediaType)
+    
+    switch authStatus {
+    case .authorized:
+      completion?(true)
+    case .denied, .restricted:
+      completion?(false)
+    case .notDetermined:
+      // prompt user for camera permission only for the first time
+      AVCaptureDevice.requestAccess(for: mediaType) { granted in
+        DispatchQueue.main.async {
+          completion?(granted)
+        }
+      }
+    @unknown default:
+      Logger.debug("Audio device in an unknown state ...")
+    }
+  }
+  
+  /// Returns current camera authorization status
+  func authorizationStatus(forType mediaType: AVMediaType) -> AVAuthorizationStatus {
+    AVCaptureDevice.authorizationStatus(for: mediaType)
+  }
```

Comment:
This method from the protocol doesn't add any true value, it is more of a middleman. 

I think it would be more beneficial if we make its protocol and the implementation public.

Let me try to explain my thought process, basically in order for the client to request access to camera permissions it needs to initialize the whole Camera object. It would be much faster if we could access the CameraService object directly and request permission from there. I don't know if a "Camera" should care if it has permissions, from my pov it only needs to be able to capture and deliver images. Concern wise I feel like they should be separated.

What do you think?

Reply:
I agree. Making `CameraService` public will have its benefits in case the user doesn't want to initialize the whole Camera object to request or check permission status.